ccip cap config contract updates

makramkd opened the pull request:
## Motivation

The most recent merge introduced breaking changes to the capability registry contract. This PR fixes them.


## Solution

* capability registry -> capabilities registry rename
* implement IERC165 for the capabilities registry
* update ICapabilitiesRegistry interface

makramkd committed:
ccip cap config contract updates

* capability registry -> capabilities registry rename
* implement IERC165 for the capabilities registry
* update ICapabilitiesRegistry interface

0xsuryansh commented on contracts/gas-snapshots/ccip.gas-snapshot:
```diff
@@ -34,56 +34,56 @@ BurnWithFromMintTokenPool_lockOrBurn:test_ChainNotAllowed_Revert() (gas: 28789)
+CCIPCapabilityConfiguration_beforeCapabilityConfigSet:test_beforeCapabilityConfigSet_CommitAndExecConfig_Success() (gas: 1844013)
+CCIPCapabilityConfiguration_beforeCapabilityConfigSet:test_beforeCapabilityConfigSet_CommitConfigOnly_Success() (gas: 1062710)
+CCIPCapabilityConfiguration_beforeCapabilityConfigSet:test_beforeCapabilityConfigSet_ExecConfigOnly_Success() (gas: 1062741)
 CCIPCapabilityConfiguration_beforeCapabilityConfigSet:test_beforeCapabilityConfigSet_OnlyCapabilityRegistryCanCall_Reverts() (gas: 9576)
```
nit: For the sake of consistency, I think we should rename the `CapabilityRegistry`, occurrences in test files as well?

https://github.com/smartcontractkit/ccip/blob/update-capconfig-after-cr-updates/contracts/src/v0.8/ccip/test/helpers/CCIPCapabilityConfigurationHelper.sol#L7 


The constructor of this helper `constructor(address capabilityRegistry)`  also has the old parameter name whereas it should be `capabilitiesRegistries` 

Let me know your thoughts

makramkd replied:
Done.

RensR commented on contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol:
```diff
@@ -3,18 +3,19 @@ pragma solidity 0.8.24;
 /// Each CR DON will have up to four configurations: for each of (commit, exec), one blue and one green configuration.
 /// This is done in order to achieve "blue-green" deployments.
-contract CCIPCapabilityConfiguration is ITypeAndVersion, ICapabilityConfiguration, OwnerIsCreator {
+contract CCIPCapabilityConfiguration is ITypeAndVersion, ICapabilityConfiguration, OwnerIsCreator, IERC165 {
```
should this also be named `capabilities` now?

makramkd replied:
That's reasonable yes.

makramkd replied:
Renamed to CCIPConfig

RensR commented on contracts/src/v0.8/ccip/capability/CCIPCapabilityConfiguration.sol:
```diff
@@ -131,9 +132,16 @@ contract CCIPCapabilityConfiguration is ITypeAndVersion, ICapabilityConfiguratio
+  function supportsInterface(bytes4 interfaceId) external pure override returns (bool) {
+    return interfaceId
+      == ICapabilityConfiguration.getCapabilityConfiguration.selector
+        ^ ICapabilityConfiguration.beforeCapabilityConfigSet.selector;
```
What's this `^` syntax?

makramkd replied:
I believe a byte-by-byte xor - this is the interface check used by the registry https://github.com/smartcontractkit/ccip/blob/9075d9b2aac6036b8a70dbbe6ef8a135574851a2/contracts/src/v0.8/keystone/CapabilitiesRegistry.sol#L1013-L1015

elatoskinas replied:
Had this question too, seems common: https://ethereum.stackexchange.com/a/40989

RensR commented on contracts/src/v0.8/ccip/capability/interfaces/ICapabilitiesRegistry.sol:
```diff
@@ -1,23 +1,31 @@
 pragma solidity ^0.8.24;
 
-interface ICapabilityRegistry {
+interface ICapabilitiesRegistry {
```
This feels like it should exist outside of ccip as it's not a ccip specific interface right?

makramkd replied:
Its not ccip specific but it is the narrow interface used by the config contract. Alternatively we can just import the full registry (they don't have an interface for it) but I'm not sure if that's preferable for compile times / contract size etc.

elatoskinas replied:
We would be able to import this after we merge in chainlink core right? And then we can remove this file

elatoskinas replied:
nvm this is the Capability Registry, not the Config interface - I recon we could add this interface under keystone, and have the keystone capability registry implement it

cc @cds95

makramkd replied:
Created CCIP-2647 to address this

makramkd committed:
rename to CCIPConfig

makramkd committed:
fmt

makramkd committed:
Merge branch 'ccip-develop' into update-capconfig-after-cr-updates

makramkd committed:
update supportsInterface impl

use type().interfaceId instead of xor expression, and add IERC165 to the
supported interfaces.

makramkd committed:
final rename
